kolibri/core/auth/csv_utils.py

```diff
@@ -0,0 +1,173 @@
+            facility = Facility.objects.get(pk=facility_id)
+        except (Facility.DoesNotExist, ValueError):
+            try:
+                facility = Facility.objects.get(name=facility_id)
+            except Facility.DoesNotExist:
+                raise ValueError(
+                    "Facility matching identifier {facility} was not found".format(
+                        facility=facility_id
+                    )
+                )
+    elif facility is not None:
+        return facility
+    else:
+        facility = Facility.get_default_facility()
+        if facility:
+            logger.info(
+                "No facility specified, using the default facility {}".format(
+                    facility.name
+                )
+            )
+        else:
+            raise ValueError(
+                "No default facility exists, please make sure to provision this device before running this command"
+            )
+    return facility
+
+
+choices_dict = dict(choices)
+
+
+def transform_choices(field, obj):
+    return choices_dict.get(obj[field], obj[field])
+
+
+mappings = {
+    "gender": partial(transform_choices, "gender"),
+    "birth_year": partial(transform_choices, "birth_year"),
+}
+
+labels = OrderedDict(
+    (
+        ("facility__name", "Facility name"),
+        ("facility__id", "Facility id"),
+        ("memberships__collection__name", "Class name"),
+        ("memberships__collection__id", "Class id"),
+        ("full_name", "Full name"),
+        ("gender", "Gender"),
+        ("birth_year", "Birth year"),
+        ("id_number", "ID number"),
+        ("username", "Username"),
+        ("password", "Password"),
+    )
+)
+
+input_fields = (
+    "full_name",
+    "username",
+    "password",
+    "facility",
+    "class",
+    "gender",
+    "birth_year",
+    "id_number",
+)
+
+
+def get_field(fields, obj):
+    for label in fields:
+        if label in obj:
+            return obj[label]
+
+
+input_choices = {val: key for key, val in choices}
+
+
+def transform_inputs(field, obj):
+    return input_choices.get(obj[field], obj[field])
+
+
+input_mappings = {
+    "class": partial(get_field, ["class", "Class id", "Class name"]),
+    "facility": partial(get_field, ["facility", "Facility id", "Facility name"]),
+}
+
+
+def map_input(obj):
+    mapped_obj = {}
+    for label in input_fields:
+        header = labels.get(label, None)
+        if label in input_mappings:
+            mapped_obj[label] = input_mappings[label](obj)
+        elif label in obj:
+            mapped_obj[label] = obj[label]
+        elif header and header in obj:
+            mapped_obj[label] = obj[header]
+    return mapped_obj
+
+
+def map_object(obj):
+    mapped_obj = {}
+    for header, label in labels.items():
+        if header in mappings and header in obj:
+            mapped_obj[label] = mappings[header](obj)
+        elif header in obj:
+            mapped_obj[label] = obj[header]
+    return mapped_obj
+
+
+db_columns = (
```

Comment:
Does this set the order? When I export, the headers come out like this (Demographic data is inserted between full name and username)

```
Facility name,Facility id,Class name,Class id,Full name,Gender,Birth year,ID number,Username,Password
```

Reply:
Also, I think the "password" field isn't being written. e.g., here's the first row of my export (data was generated using the UN Women CM executable during a hack session).

```
Facility name,Facility id,Class name,Class id,Full name,Gender,Birth year,ID number,Username,Password
a,6f963c47fefcfbe2d1aaec8490dba5da,,,j o r d a n,,,,jordan,
```

Reply:
The `labels` OrderedDict defines the order, so can shuffle that to match.

The password can't be exported because we only store the hashed value. The only reason for including it here is to have the column header available in case someone wants to change the password.